frontend/document/document.lua

```diff
@@ -120,13 +120,18 @@ end
 -- Note that if PDF file size is around 1024, 4096, 16384, 65536, 262144
 -- 1048576, 4194304, 16777216, 67108864, 268435456 or 1073741824, appending data
 -- by highlighting in KOReader may change the digest value.
-function Document:fastDigest()
+function Document:fastDigest(docsettings)
     if not self.file then return end
     local file = io.open(self.file, 'rb')
     if file then
-        local docsettings = require("docsettings"):open(self.file)
+        local own_docsettings = false
```

Comment:
own_docsettings? why not `has_docsettings`? :-P

Reply:
In this context, `has_docsettings` is not a good name.
It's when this function is not provided with an already loaded docsetting object (as function argument) that it will itself open its _own_ docsettings, and if it does, this flag is set to true, so it can remember it should close/flush it.

Reply:
`local_docsettings`? `tmp_docsettings`? `_docsettings`? I don't know, I just don't think "own" is particularly clear. :-P

Reply:
ok, let's go with `tmp_docsettings` (because `local local_docsettings = false` looks looks bad bad :)